### PR TITLE
[Keyboard] Add Marcopad

### PR DESCRIPTION
**Pull request by marcopelegrini**
## Description

Add Marcopad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/marcopad/readme.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Marcopad
+
+![macropad](imgur.com image replace me!)
+
+*A compact 3x3 keys macropad built with RP-2040 Zero*
+
+* Keyboard Maintainer: [Marco Pelegrini](https://pelegrini.ca/marcopad)
+* Hardware Supported: [Macropad PCB](https://pelegrini.ca/marcopad/hardware)
+* Hardware Availability: [Macropad](https://pelegrini.ca/marcopad)
```
Comment (zvecr) [inline keyboards/marcopad/readme.md:9]: These links are all dead

File: keyboards/marcopad/keyboard.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	},
+	"url": "https://pelegrini.ca/marcopad",
+	"usb": {
+		"device_version": "1.0.0",
+		"pid": "0x0000",
+		"vid": "0xFEED"
+	},
+	"backlight": {
+		"pin": "GP28"
+	},
```
Comment (zvecr) [inline keyboards/marcopad/keyboard.json:27]:
```suggestion
	"ws2812": {
        "driver": "vendor",
		"pin": "GP28"
	},
```

File: keyboards/marcopad/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        KC_P7,   KC_P8,   KC_P9,
+        KC_P4,   KC_P5,   KC_P6,
+        KC_P1,   KC_P2,   KC_P3
+    )
+};
+
+const uint16_t PROGMEM backlight_combo[] = {KC_P7, KC_P8, COMBO_END};
+combo_t key_combos[] = {
+    COMBO(backlight_combo, UG_NEXT)
+};
```
Comment (zvecr) [inline keyboards/marcopad/keymaps/default/keymap.c:26]: > * default keymaps should be "pristine"
>   * what does pristine mean? no custom keycodes. no advanced features like tap dance or macros

```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/marcopad/keyboard.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+			"knight": true,
+			"christmas": true,
+			"static_gradient": true,
+			"rgb_test": true,
+			"alternating": true,
+			"twinkle": true
+		}
+	},
+	"layouts": {
+		"LAYOUT": {
```
Comment (dunk2k) [inline keyboards/marcopad/keyboard.json:46]:
```suggestion
		"LAYOUT_ortho_3x3": {
```

File: keyboards/marcopad/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    /*
+     * ┌───┬───┬───┐
+     * │ 7 │ 8 │ 9 │
+     * ├───┼───┼───┤
+     * │ 4 │ 5 │ 6 │
+     * ├───┼───┼───┤
+     * │ 1 │ 2 │ 3 │
+     * └───┴───┴───┘
+     */
+    [0] = LAYOUT(
```
Comment (dunk2k) [inline keyboards/marcopad/keymaps/default/keymap.c:16]:
```suggestion
    [0] = LAYOUT_ortho_3x3(
```

File: keyboards/marcopad/keyboard.json [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+			"rainbow_swirl": true,
+			"snake": true,
+			"knight": true,
+			"christmas": true,
+			"static_gradient": true,
+			"rgb_test": true,
+			"alternating": true,
+			"twinkle": true
+		}
+	},
```
Comment (dunk2k) [inline keyboards/marcopad/keyboard.json:44]:
```suggestion
	},
	"community_layouts": ["ortho_3x3"],
```

**Review by dunk2k — CHANGES_REQUESTED** [review]
Add Community Layout support

File: keyboards/marcopad/readme.md (hunk 1 above)
Comment (marcopelegrini) [reply keyboards/marcopad/readme.md:9]: I'm publishing the site at the moment, chicken and egg situation. Will close this when done

File: keyboards/marcopad/keyboard.json [hunk 7]
```diff
@@ -0,0 +1,58 @@
+{
+	"manufacturer": "pelegrini.ca",
+	"keyboard_name": "marcopad",
+	"maintainer": "Marco Pelegrini",
+	"development_board": "promicro_rp2040",
+	"diode_direction": "ROW2COL",
+	"features": {
+		"bootmagic": true,
+		"rgblight": false
```
Comment (dunk2k) [inline keyboards/marcopad/keyboard.json:9]: Rgblight feature is disabled by default 
```suggestion
```

File: keyboards/marcopad/keyboard.json [hunk 8]
```diff
@@ -0,0 +1,58 @@
+{
+	"manufacturer": "pelegrini.ca",
+	"keyboard_name": "marcopad",
+	"maintainer": "Marco Pelegrini",
+	"development_board": "promicro_rp2040",
```
Comment (dunk2k) [inline keyboards/marcopad/keyboard.json:5]: Could `development_board` be replaced by `processor` and `bootloader'? As the WaveShare RP2040 Zero has a different physical footprint and different peripheral configuration from the SparkFun Pro Micro RP2040 🙈

File: keyboards/marcopad/keymaps/backlit/rules.mk [hunk 9]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2022 Marco Pelegrini <marco@pelegrini.ca>
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+RGBLIGHT_ENABLE = yes
+COMBO_ENABLE = yes
```
Comment (dunk2k) [inline keyboards/marcopad/keymaps/backlit/rules.mk:5]:
```suggestion
COMBO_ENABLE = yes

```

File: keyboards/marcopad/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+     * ├───┼───┼───┤
+     * │ 1 │ 2 │ 3 │
+     * └───┴───┴───┘
+     */
+    [0] = LAYOUT_ortho_3x3(
+        KC_P7,   KC_P8,   KC_P9,
+        KC_P4,   KC_P5,   KC_P6,
+        KC_P1,   KC_P2,   KC_P3
+    )
+};
```
Comment (dunk2k) [inline keyboards/marcopad/keymaps/default/keymap.c:21]:
```suggestion
};

```

File: keyboards/marcopad/keymaps/backlit/keymap.c [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+        KC_P7,   KC_P8,   KC_P9,
+        KC_P4,   KC_P5,   KC_P6,
+        KC_P1,   KC_P2,   KC_P3
+    )
+};
+
+const uint16_t PROGMEM backlight_combo[] = {KC_P7, KC_P8, COMBO_END};
+combo_t key_combos[] = {
+    COMBO(backlight_combo, UG_NEXT)
+};
```
Comment (dunk2k) [inline keyboards/marcopad/keymaps/backlit/keymap.c:26]:
```suggestion
};

```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/marcopad/keyboard.json
Comment (waffle87) [inline keyboards/marcopad/keyboard.json:1]: Please run `qmk format-json` on this file.

File: keyboards/marcopad/keyboard.json [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+	"keyboard_name": "MarcoPad",
+	"manufacturer": "pelegrini.ca",
+	"url": "https://pelegrini.ca/marcopad",
+	"maintainer": "Marco Pelegrini",
+	"bootloader": "rp2040",
+	"processor": "RP2040",
+	"diode_direction": "ROW2COL",
+	"features": {
+		"bootmagic": true,
+		"rgb_matrix": true
```
Comment (waffle87) [inline keyboards/marcopad/keyboard.json:11]:
```suggestion
        "rgb_matrix": true,
        "mousekey": true,
        "extrakey": true
```

File: keyboards/marcopad/readme.md [hunk 13]
```diff
@@ -0,0 +1,27 @@
+# MarcoPad
+
+![marcopad](https://pelegrini.ca/assets/img/marcopad/hardware/macropad.png)
```
Comment (waffle87) [inline keyboards/marcopad/readme.md:3]: This image should be rehosted on Imgur.

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**